File: src/core/java/src/net/lintim/util/SolverType.java
package net.lintim.util;

/**
 * Enum for used solver types in LinTim
 */
public enum SolverType {
    XPRESS("Xpress"),
    GUROBI("Gurobi"),
    CPLEX("Cplex"),
    GLPK("Glpk");

    private String name;

    /**
     * Initialize a solver type with the given name
     * @param name the name of the solver
     */
    SolverType(String name){
        this.name = name;
    }

    @Override
    public String toString(){
        return this.name;
    }
}

File: src/core/python/core/util/solver_type.py
from enum import Enum


class SolverType(Enum):
    """
    Enum for used solver types in LinTim
    """
    XPRESS = "XPRESS"
    GUROBI = "GUROBI"
    CPLEX = "CPLEX"
    GLPK = "GLPK"
